integration-test-groups/foundation/scheduler/src/test/java/org/apache/camel/quarkus/component/scheduler/it/SchedulerTest.java

```diff
@@ -28,12 +28,42 @@
 class SchedulerTest {
 
     @Test
-    public void test() throws Exception {
-        // wait until the scheduler has run and return a counter that is > 0
+    public void testInitialDelay() throws Exception {
         await().atMost(5, TimeUnit.SECONDS).until(() -> {
             String body = RestAssured.get("/scheduler/get").then().statusCode(200).extract().body().asString();
             return !body.equals("0");
         });
     }
 
+    @Test
+    public void testDelay() throws Exception {
+
+        await().atMost(2, TimeUnit.SECONDS).until(() -> {
```

Comment:
Could the body could be > 1 even if withDelay is never called ?
At the end of the day, we could maybe better to separate `withDelay` and `useFixedDelay`.
So, 2 distinct tests and 2 distinct atomic integers.
What do you think ?

Reply:
@aldettinger do you mean two separate Atomic int like myCounter or its just during testing two separate int one for withDelay and one for withFixedDelay ?

Reply:
The main idea is that, from my point of view, the current implementation of this test could pass even when the `withDelay` route is never called.
From there, one solution could be to have two distinct AtomicInteger, let's say `withDelayCounter` and `useFixedDelayCounter`. And then, it's possible to check that each route has been invoked separately.